packages/ui-components/src/Avatar/Avatar.tsx

```diff
@@ -14,6 +13,13 @@ export interface AvatarProps {
   transformCase?: AvatarCase;
 }
 
+type NativeDivProps = Omit<
```

Comment:
It doesn't look like any HTML attributes will be passed to the child div so I don't think this type is accurate.

Reply:
Do you mean that since `OwnAvatarProps` doesn't overlap any of the `HTMLAttributes` (which is true) and that's why it's not accurate? Or do you mean that `React.HTMLAttributes<HTMLDivElement>` doesn't return any props? I'm adding this to use `className` and `children` from the underlying div prop types instead of defining them explicitly for Avatar.

Reply:
>I'm adding this to use className and children from the underlying div prop types instead of defining them explicitly for Avatar.

`children` is included with `React.FC` so this is just adding `className`.  

>Do you mean that since OwnAvatarProps doesn't overlap any of the HTMLAttributes (which is true) and that's why it's not accurate? 

Doesn't `onClick` overlap? I thought that was why `Omit` was being used. Why is Omit being used if there are no overlapping props?

`React.HTMLAttributes<HTMLDivElement>` includes a lot more than `className` which is why I don't think it's accurate. 

> Constructs a type by picking all properties from Type and then removing Keys. - [TS docs](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys)

```
Omit<
  React.HTMLAttributes<HTMLDivElement>, // all HTMLDivElement props
  keyof OwnAvatarProps // removing any overlapping keys
>;
```
Am I missing something?

Reply:
Perhaps you are right. I was copying this pattern from other components because I thought this was a solved problem. Here's _what I want_,

1) `OwnAvatarProps` to be props we have defined to provide functionality of our component
2) `NativeDivProps` to be valid props of the underlying `<div>` element that will be rendered as the Avatar
3) `AvatarProps` to be the union type of both of these so that the user of `<Avatar>` can add any prop to the component either defined by our documentation or any valid prop of a `<div>`.

My expectation is that when using an `Avatar` that it should honor any prop that is valid html attribute for the underlying element. So that I can add a `className` or a data attribute or an `onClick`.

So I take it back, you are totally right, the `Omit` in this case is overriding the `onClick` from `React.HTMLAttributes<HTMLDivElement>` in favor our `onClick` prop.

But what is missing, that I think you are calling out, is that I want to pass along all of the rest of the props (besides className) that they user may have defined to the underlying `div`, like this

```jsx
const Avatar: React.FC<AvatarProps> = ({
  children,
  className,
  intent = "default",
  size = "default",
  disabled = false,
  selectable = false,
  onClick,
  transformCase = "uppercase",
  ...rest
}) => {
 {/*      snip          */}
   
 return (
    <div className={classnames} onClick={onClickHandler} {...rest}>
      {children}
    </div>
  );
};
```